### PR TITLE
Preserve WebXR light estimation on WebGPU

### LINKED ISSUE #18770: [WebGPU-XR][Phase 4C] Preserve light estimation without WebGPU reflection cube maps
Part of #18636 and #18635. Depends on Phase 4A.

We only accept feature requests that are discussed on the BabylonJS forum

N/A — this work is already tracked and discussed under the WebGPU-for-WebXR epic #18635.

**Is your feature request related to a problem? Please describe.**
WebXR Light Estimation mixes graphics-neutral light-probe data with a reflection cube-map path that constructs `XRWebGLBinding`. Disabling the whole feature on WebGPU would unnecessarily discard direction, intensity, and spherical harmonics.

**Describe the solution you'd like**
- Preserve `requestLightProbe` and frame light estimates on WebGPU XR.
- Continue updating light direction, intensity, and spherical harmonics.
- If reflection cube maps were requested, disable only that output with one specific warning because `XRGPUBinding` has no equivalent.
- Do not construct a WebGL binding or wrap WebGL textures in a WebGPU session.
- Preserve existing WebGL reflection options, listener lifecycle, polling, prefiltering, and cache reset behavior.

Acceptance: deterministic tests prove graphics-neutral estimates continue while reflection stays unavailable on WebGPU; WebGL reflection behavior has negative controls and remains unchanged.

**Discussion**
Tracked architecture discussion: #18635 and #18636.

### PR DESCRIPTION
**Pull request by RaananW**
> 🤖 *This PR was created by the create-pr skill.*

## Summary

- Preserve graphics-neutral WebXR light direction, intensity, and spherical harmonics estimates on WebGPU.
- Disable only XRWebGLBinding reflection cube maps on WebGPU, with one specific warning when reflection was requested.
- Reuse the cached Phase 4A WebGL graphics binding while preserving existing reflection options and lifecycle behavior.
- Add deterministic WebGPU and WebGL regression coverage.

## Validation

- Core XR unit tests: 22 files, 341 tests passed.
- Core TypeScript compilation passed.
- Targeted formatting and lint passed with no errors.
- Tree-shaking and side-effect synchronization checks passed.

The Quest runtime used for manual validation did not expose Light Estimation in either WebGPU or WebGL2, so device validation was inconclusive rather than failed. XR entry itself succeeded.

Closes #18770

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by RaananW** [comment]
Self code review completed by the code-review skill. No issues found.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/XR/features/WebXRLightEstimation.pure.ts [hunk 1]
```diff
@@ -317,15 +322,12 @@ export class WebXRLightEstimation extends WebXRAbstractFeature {
     public override detach(): boolean {
         const detached = super.detach();
 
-        if (this._xrLightProbe !== null && !this.options.disableCubeMapReflection) {
+        if (this._xrLightProbe !== null && this._reflectionCubeMapEnabled) {
             this._xrLightProbe.removeEventListener("reflectionchange", this._updateReflectionCubeMap);
             this._xrLightProbe = null;
         }
```
[the remark this answers is not in the document]
Comment (RaananW) [reply packages/dev/core/src/XR/features/WebXRLightEstimation.pure.ts]: Fixed in dfe5ec8b5b. `detach()` now clears `_xrLightProbe` for every configuration while keeping listener removal conditional on reflection cube maps. I also added lifecycle assertions for WebGPU and caller-disabled reflection.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/18837/merge

Test environment:
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18837/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18837/merge/index.html#WGZLGJ#4600

Links to test your changes to core in the published versions of the Babylon tools (does not contain changes you made to the tools themselves):

https://playground.babylonjs.com/?snapshot=refs/pull/18837/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/18837/merge
https://gui.babylonjs.com/?snapshot=refs/pull/18837/merge
https://nme.babylonjs.com/?snapshot=refs/pull/18837/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/18837/merge#BCU1XR#0

*If you made changes to the sandbox or playground in this PR, additional comments will be generated soon containing links to the dev versions of those tools.*

**Comment by bjsplat** [comment]
Interaction tests

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18837/merge/testResults/interactionplaywright/index.html

**Comment by bjsplat** [comment]
## ⚡ Performance Test Results

🟢 All performance tests passed — no regressions detected.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18837/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18837/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
## 🟢 Memory Leak Test Results

**4** passed, **0** leaked out of **4** scenarios

🟢 All memory leak tests passed — no leaks detected.

<details>
<summary>Passed Scenarios (4)</summary>

| Scenario | Package |
|----------|---------|
| Core Playground #2FDQT5#1508 | @babylonjs/core |
| Core Playground #T90MQ4#14 | @babylonjs/core |
| Core Playground #8EDB5N#2 | @babylonjs/core |
| Core Playground #LL5BIQ#636 | @babylonjs/core |

</details>

**Review by Popov72 — APPROVED** [review]
